packages/block-library/src/search/view.js

```diff
@@ -66,6 +81,41 @@ const { actions } = store(
 					actions.closeSearchInput();
 				}
 			},
+			*updateSearch() {
+				const { ref } = getElement();
+				const { value } = ref;
+
+				// Don't navigate if the search didn't really change.
+				if ( value === state.search ) {
+					return;
+				}
+
+				state.search = value;
+
+				// Debounce the search by 300ms to prevent multiple navigations.
+				// We can do this by yielding a promise that resolves after 300ms and
+				// then bailing out if the search has changed.
+				yield new Promise( ( resolve ) => setTimeout( resolve, 300 ) );
+				if ( value !== state.search ) {
+					return;
+				}
+
+				const url = new URL( window.location );
```

Comment:
Minor: the URL constructor takes a string or URL object. This will work by coercing `window.location` to a string, but passing `window.location.href` correctly passes the string:

```suggestion
				const url = new URL( window.location.href );
```

Reply:
done in aaeee09cbdaba188bd2c5aa926f0a883bc7dbe56